app/src/main/java/com/example/reflector_android/MainActivity.kt

```diff
@@ -43,4 +59,55 @@ open class MainActivity : AppCompatActivity() {
             else -> super.onOptionsItemSelected(item)
         }
     }
+
+    private fun addScrollerListener(articls: MutableList<com.example.reflector_android.network.Article>?){
+        RecyclerView.addOnScrollListener(object: RecyclerView.OnScrollListener(){
+            @RequiresApi(Build.VERSION_CODES.O)
+            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                if (!isLoading){
+                    if (layoutManager.findLastCompletelyVisibleItemPosition() == articls?.size?.minus(1)){
+                        loadMore(articls)
+                        isLoading = true
+                    }
+                }
+            }
+        })
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun loadMore(articlesList: MutableList<com.example.reflector_android.network.Article>?){
+        var date: LocalDate? = null
+        RecyclerView.post(Runnable {
+            var nullArticle = com.example.reflector_android.network.Article("", "", date , "", "", isLoading)
+            articlesList?.add(nullArticle)
+            adapter.notifyItemInserted(articlesList?.size?.minus(1)!!)
+        })
+
+        RecyclerView.postDelayed(Runnable {
+            var loadingRemoved = articlesList?.size?.minus(1)
+            GlobalScope.launch {
+                val service = async { RSSService().fetchMoreNews(articlesList?.size!!)}
+                moreArticles = service.await()
+                if (loadingRemoved != null) {
+                    runOnUiThread {
+                        articlesList?.removeAt(loadingRemoved)
+                    }
+                }
+                if (articlesList != null) {
+                    runOnUiThread {
+                        adapter.notifyItemRemoved(articlesList.size)
+                    }
+                }
+                for (i in moreArticles!!){
+                    runOnUiThread {
+                        articlesList?.add(i)
+                        adapter.notifyItemInserted(articlesList?.size?.minus(1)!!)
+                    }
+                }
+                isLoading = false
+            }
+
+        }, 2500)
```

Comment:
`2500` is a magic number. Make a `private` variable that defines what `2500` is.

https://flaviocopes.com/magic-numbers/